OAK-11172: add documentation about how to run DB2 in Docker

mbaedke opened the pull request.

mbaedke committed:
OAK-11172: add documentation about how to run DB2 in Docker

Updated DB2 driver dependency.

mbaedke committed:
OAK-11172: add documentation about how to run DB2 in Docker

Updated DB2 driver dependency.

reschke reviewed (changes requested):
Could you also please add the source of the docker file to oak-store-document/src/test/resources.

Furthermore please update [oak-doc](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-doc/src/site/markdown/testing.md) as well.

**DONE**: https://github.com/apache/jackrabbit-oak/pull/1774/commits/07254e46274bbc3c9e1a5e0e7a583e304e4494c6

reschke committed:
OAK-11172: DB2 support - move JDBC URL change

reschke reviewed (changes requested):
We also should update the version checks in https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/rdb/RDBDocumentStoreDB.java

**(DONE)**

mbaedke committed:
Changed DB name from oak back to OAK

reschke committed:
OAK-11172: updated expected DB and JDBC versions for DB2

reschke committed:
OAK-11172: add DB2 docker example to documentation

reschke reviewed (changes requested):
I think we are done once the docker source file is added.

mbaedke committed:
OAK-11172: add documentation about how to run DB2 in Docker

Added Dockerfile to build an image with a pre-installed database ready to be used with Oak.

mbaedke committed:
OAK-11172: add documentation about how to run DB2 in Docker

Linked Dockerfile to documentation.

mbaedke committed:
Fixed typo in documentation

reschke reviewed (changes requested):
"Dockerfile" needs a more specific name, or needs to move into a more specific directory (such as test/resources/rdb/db2/Dockerfile).

mbaedke committed:
OAK-11172: add documentation about how to run DB2 in Docker

Moved Dockerfile to a more reasonable location.

mbaedke committed:
Fixed link in documentation.

reschke commented on oak-doc/src/site/markdown/testing.md:
```diff
@@ -70,3 +72,17 @@ This pulls the docker image "postgres:13-alpine", specifies a system password an
+
+`docker run -h db2server --name db2server --privileged=true -p 50000:50000 baedke/db2-community-oak:11.5.9.0`
+
+The Docker image baedke/db2-community-oak:11.5.9.0 may also be build from the official DB2 community image `icr.io/db2_community/db2:11.5.9.0` using the Dockerfile https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/test/resources/db2/Dockerfile
```
```suggestion
The Docker image baedke/db2-community-oak:11.5.9.0 may also be built from the official DB2 community image `icr.io/db2_community/db2:11.5.9.0` using the Dockerfile https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/test/resources/db2/Dockerfile.
```

mbaedke committed:
Fixed typo in documentation

Co-authored-by: Julian Reschke <reschke@apache.org>
